Update for CUDA 13.0.0

billysuh7 opened the pull request:
xref: https://github.com/conda-forge/cuda-feedstock/issues/85

billysuh7 committed:
Update for CUDA 13.0.0

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found some lint.

Here's what I've got...


For **recipe/meta.yaml**:

 * ❌ Selectors are suggested to take a ``<two spaces>#<one space>[<expression>]`` form. See lines [278]


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/16733972265](https://github.com/conda-forge/conda-forge-webservices/actions/runs/16733972265). Examine the logs at this URL for more detail.</sub>

billysuh7 commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 25.7.0, conda-smithy 3.51.1, and conda-forge-pinning 2025.08.04.10.37.43

billysuh7 committed:
fix lint; update tests
Forgot to apply nvcc profile patch for linux
forgot to apply processing of multiple sources
Forgot to copy right meta.yaml
update version, turn off tegra
update sha256sum; merge https://github.com/conda-forge/cuda-nvcc-impl-feedstock/pull/48
fix windows build script; fix typo

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

billysuh7 committed:
no need to guard stdlib c

billysuh7 commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 25.7.0, conda-smithy 3.51.1, and conda-forge-pinning 2025.08.04.23.25.35

billysuh7 commented:
Verified the package contents with ones from nvidia channel to make sure nothing is missed.

bdice commented on recipe/meta.yaml:
```diff
@@ -118,27 +138,25 @@ outputs:
         - arm-variant * {{ arm_variant_type }}  # [aarch64]
         - cuda-version {{ cuda_version }}
-        - libgcc {{ gcc_min_constraint }}  # [linux]
+        - libgcc {{ gcc_min_constraint }}  # [linux64]
```
We need this constraint on `aarch64` too, right? Should this be reverted to `# [linux]`?

billysuh7 replied:
I totally agree, thanks for catching!

bdice commented on recipe/meta.yaml:
```diff
@@ -103,7 +123,7 @@ outputs:
         - cuda-version >={{ cuda_version }},<{{ cuda_version_next_major }}
     files:
-      - targets/{{ target_name }}/bin/nvcc                          # [linux]
+      - targets/{{ target_name }}/bin/nvcc
```
Why was the `# [linux]` selector removed here?

billysuh7 replied:
This was purely a mistake. Thanks for catching again.

bdice commented on recipe/meta.yaml:
```diff
@@ -1,11 +1,9 @@
 {% set gcc_constraint = ">=6,<15.0a0" %}
```
GCC 15 support was added in CUDA 13. https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html

> New host compiler support in CUDA 13.0:
> - Clang 20
> - GCC 15

```suggestion
{% set gcc_constraint = ">=6,<16.0a0" %}
```

billysuh7 committed:
fix typos

billysuh7 committed:
Update recipe/meta.yaml

Co-authored-by: Bradley Dice <bdice@bradleydice.com>

bdice commented on recipe/meta.yaml:
```diff
@@ -125,20 +145,18 @@ outputs:
         - {{ pin_subpackage("libnvptxcompiler-dev_" + target_platform, exact=True) }}
         - arm-variant * {{ arm_variant_type }}    # [aarch64]
-        - libgcc {{ gcc_min_constraint }}  # [linux]
+        - libgcc {{ gcc_min_constraint }}  # [linux64]
```
```suggestion
        - libgcc {{ gcc_min_constraint }}  # [linux]
```

bdice committed:
linux instead of linux64

robertmaynard reviewed (approved):
nvcc.profile patch looks correct to me
